### PR TITLE
User agent for Python (and formerly Ruby, already merged)

### PR DESCRIPTION
**Pull request by sah**
Addresses #5657 for Python and Ruby

- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

**Comment by p0deje** [comment]
Can you explain why you need these changes?

**Comment by sah** [comment]
Explanation is in the issue referenced in the description, but basically this is useful for error diagnosis in cloud services and grids. Similar changes have already been introduced for Java and .NET, so this is just rolling the feature out to more languages.

**Comment by p0deje** [comment]
Thank you, I've merged Ruby part in 429af62c79c4048d54856a9d4b79d451d98e372a!

**Comment by sah** [comment]
Updated this PR to reflect just the Python change now that the Ruby change has been merged.

File: py/selenium/webdriver/remote/remote_connection.py [hunk 1]
```diff
@@ -172,10 +174,14 @@ def get_remote_connection_headers(cls, parsed_url, keep_alive=False):
...
 
+        system = platform.system().lower()
+        if system == "darwin":
+            system = "mac"
+
         headers = {
             'Accept': 'application/json',
             'Content-Type': 'application/json;charset=UTF-8',
-            'User-Agent': 'Python http auth'
+            'User-Agent': 'selenium/%s (python %s)' % (__version__, system)
```
Comment (lmtierney) [inline py/selenium/webdriver/remote/remote_connection.py]: Prefer `str.format` for String Interpolation in the future

File: py/test/unit/selenium/webdriver/remote/test_remote_connection.py [hunk 2]
```diff
@@ -33,7 +34,8 @@ def test_get_remote_connection_headers_defaults():
     assert 'Connection' not in headers.keys()
     assert headers.get('Accept') == 'application/json'
     assert headers.get('Content-Type') == 'application/json;charset=UTF-8'
-    assert headers.get('User-Agent') == 'Python http auth'
+    assert headers.get('User-Agent').startswith("selenium/%s (python " % __version__)
+    assert headers.get('User-Agent').split(' ')[-1] in {'windows)', 'mac)', 'linux)'}
```
Comment (lmtierney) [inline py/test/unit/selenium/webdriver/remote/test_remote_connection.py]: Is there a reason you aren't just asserting the entire `User-Agent`?  The second assertion could allow false positives if the wrong platform sent

Comment (sah) [reply py/test/unit/selenium/webdriver/remote/test_remote_connection.py]: My reasoning was that asserting the whole thing requires duplicating the code to calculate the platform name, which proves nothing except that that code does whatever it does. This way if on OS X it returns "darwin" or Windows it returns "win32", that issue will be caught when these tests are run on those platforms. I don't feel strongly about it though, happy to go a different route here if you prefer.

**Pull request merged**